libbeat/outputs/elasticsearch/api.go

```diff
@@ -59,6 +60,26 @@ type Total struct {
 	Relation string `json:"relation"`
 }
 
+// UnmarshalJSON correctly unmarshal the hits response between ES 6.0 and ES 7.0.
+func (t *Total) UnmarshalJSON(b []byte) error {
+	// fallback for Elasticsearch < 7
+	if i, err := strconv.Atoi(string(b)); err == nil {
+		*t = Total{Value: i, Relation: "eq"}
+		return nil
+	}
+
+	value := struct {
+		Value    int    `json:"value"`
+		Relation string `json:"relation"`
+	}{}
+
+	if err := json.Unmarshal(b, &value); err != nil {
```

Comment:
How about doing this first and falling back to the < 7 version if it fails, since it's more likely we're talking to es 7.x from beats 7.x.